src/main/java/seedu/address/model/room/RoomList.java

```diff
@@ -14,28 +13,39 @@
 /**
  * Contains information regarding the Room information
  */
-public class RoomList {
+public class RoomList implements ReadOnlyRoomList {
     private static final Logger logger = LogsCenter.getLogger(JsonAddressBookStorage.class);
 
     private int numOfRooms;
     private PriorityQueue<Room> rooms = new PriorityQueue<>();
-    private Room[] roomsInArray = new Room[0];
     private ObservableList<Room> roomObservableList = FXCollections.observableArrayList();
 
     /** Creates default RoomList() object where all fields are null**/
     public RoomList() {}
 
+    /**
+     * Converts data from readOnlyRoomList to roomList
+     */
+    public RoomList(ReadOnlyRoomList readOnlyRoomList) {
+        this();
+        resetData(readOnlyRoomList);
+    }
     /**
      * Creates a RoomList object using the information given in files containing information about
      * which rooms are occupied and number of rooms
      */
-    public RoomList(PriorityQueue<Room> rooms, Room[] roomsInArray, int numOfRooms) {
+    public RoomList(PriorityQueue<Room> rooms, int numOfRooms) {
         this.rooms = rooms;
-        this.roomsInArray = roomsInArray;
         this.numOfRooms = numOfRooms;
         convertPriorityQueue(rooms);
```

Comment:
i think you can delete this line and subsequently, the converyPriorityQueue method